lib/rules/a11y-no-visually-hidden-interactive-element.js

```diff
@@ -0,0 +1,90 @@
+const {getProp, getPropValue} = require('jsx-ast-utils')
+const {getElementType} = require('../utils/get-element-type')
+const {generateObjSchema} = require('eslint-plugin-jsx-a11y/lib/util/schemas')
+
+const defaultClassName = 'sr-only'
+const defaultcomponentName = 'VisuallyHidden'
+
+const schema = generateObjSchema({
+  className: {type: 'string'},
+  componentName: {type: 'string'},
+})
+
+/** Note: we are not including input elements at this time
+ * because a visually hidden input field might cause a false positive.
+ * (e.g. fileUpload https://github.com/primer/react/pull/3492)
+ */
+const INTERACTIVELEMENTS = ['a', 'button', 'summary', 'select', 'option', 'textarea']
+
+const checkIfInteractiveElement = (context, node) => {
+  const elementType = getElementType(context, node.openingElement)
+  const asProp = getPropValue(getProp(node.openingElement.attributes, 'as'))
```

Comment:
I love this approach of just directly grabbing the tag from `as`, a lot more than this experimental approach I added some time ago! I've been wanting to revisit and remove this - [Component mapping experimental](https://github.com/github/eslint-plugin-github#component-mapping-experimental)! 

Since `as` is Primer React specific, can we allow the `as` to be configured in the config as well? (e.g. maybe some design systems use `tag` or `component`). 

I think this line can be moved into `getElementType`, and we can get rid of the comopnent mappign stuff (lines 15-34)!

Reply:
Maybe we can pair on this in our sync :)

Reply:
`htmlPropName` - name for 'as' config